wagtail/documents/views/documents.py

```diff
@@ -271,15 +271,23 @@ def usage(request, document_id):
         raise PermissionDenied
 
     paginator = Paginator(doc.get_usage(), per_page=20)
-    used_by = paginator.get_page(request.GET.get("p"))
+    object_page = paginator.get_page(request.GET.get("p"))
 
     # Add edit URLs to each source object
     url_finder = AdminURLFinder(request.user)
-    for object, references in used_by:
-        object.edit_url = url_finder.get_edit_url(object)
+    results = []
+    for object, references in object_page:
+        edit_url = url_finder.get_edit_url(object)
+        if edit_url is None:
+            label = _("(Private %s)") % object._meta.verbose_name
+            edit_link_title = None
+        else:
+            label = str(object)
+            edit_link_title = _("Edit this %s") % object._meta.verbose_name
```

Comment:
Wondering if it's possible to word this in a way to make it easier to translate for languages with genders? (we've probably got a lot of cases like this so maybe this should be solved separately)

Reply:
Yep, this is already a frequent issue in the Wagtail codebase, particularly within modeladmin and snippets. #7117 raises this and a couple of similar issues.